### ISSUE TITLE
Enhance locations (on page)  where content is displayed

### ISSUE THREAD

**Issue by cjcornell** [PropelVentures/GenericPlatform#57]
### Enhance locations (on page)  where content is displayed
- use tab_num   (new parameters) to direct GPC to display content on different areas of page
- particularly:
   - right column, left column  (sidebars)
   - "top" area  (above tabbed content)
   - "bottom area" (below tabbed content)
   - several "center" areas (instead of tabbed content)
- detect tab_num such as 
   - R1, R2 ...  L1, L2   to indicate left/right columns/side ...
      - this will, for instance, allow the profile photo to appear there
   - C1, C2 C3 ... 
      - this is special because it is used *instead* of tabs (sort of)
      - content is still displayed simular to tabs each one is "serial" under the other  (so you can scroll down th page and see each content area.
      - this will probably only be used for lists (or display-only content) to avoid conflects with multiple "update"/submit buttons
- Note ... one of the main purpose(s) of this  is:
   - to display sidebars - for Extra Menus/Links and extra content (ad content???)
  -  to display the profile photo or other info in a distinct and controllable area (ie we can use DD to specify)
   - to display additional formatted content above or below tabbed content areas.

**Comment by yasirkhan177** [comment]
several "center" areas (instead of tabbed content)  <<< this mean if we define C1, C2 & C3 ..  this will display content of C1 first and than will display contents of C2 (under C1 ) and so on ..

**Comment by yasirkhan177** [comment]
the above whole layout is for one singular page . so i think instead of using DD->tab_num .. we should use any field( new field ) in navigation table . because navigation table define more on the top and it is used for one whole page means for dd->display_page ..

**Comment by cjcornell** [comment]
No .... Because the nav table is more about defining menus.
But I am not exactly sure what you are proposing but using the nav table for this does not make sense for the way I envision using tabnum = 0

**Comment by yasirkhan177** [comment]
its not only about tab_num=0 .. its about handling the layout of the page by introducing new field in navigation table .. for example .. for dd->display_page='myaccount' <<< there is one entry in the navigation table .. and for dd->display_page='myprojects' << there is another entry in the navigation table .. so from navigation table .. we can tell GPC .. how to display the content and layout of the page . what you say?

**Comment by yasirkhan177** [comment]
there can be another scenario where you don't want to introduce a top level layout for the page .. Means if you want to have different layout for each tab .. then we shall go with tab_num .. but i don't see the approach as flexible as i stated. i don't have problem with going with your approach .. i'm only discussing it for better result.

**Comment by cjcornell** [comment]
- before this gets coded, you will need to explain to me how this will work in a little more detail (or more clearly) - because I cannot quite see how your proposed approach achieves my needs without extra complications, and
- "a different layout for each tab" ... for the short or even long term future, I don't really see doing this ... so if this is your reason for the new structure, it is not necessary.

**Comment by yasirkhan177** [comment]
After my investigation .. i decided to go with tab_num .... put some entries in database  DD table .. for R1 ,R2 , L1,L2  ETC . and let me know

**Comment by cjcornell** [comment]
OK - sounds good.
- After you code, test by making some entries on the HOME page ....  (starting after the banner)
- Those entries being Box lists in the center, and maybe something in the right/left sidebars
- We forgot one logical other "area"  (B1, B2 ...) which imply "bottom" areas .. though I am not sure this makes much sense.

**Comment by yasirkhan177** [comment]
right now coding for profile page

**Comment by cjcornell** [comment]
Profile page = almost **any** page ...  (except home), correct?

**Comment by yasirkhan177** [comment]
yes

**Comment by yasirkhan177** [comment]
yes correct... you didn't answer to my question .. for sidebars ... same formating function doesn't make any sense .. because in sidebars(left or right or both ) we are not going to display any fields to update or lists .. there will be different things to display .. so different functions will be used.. correct me if i'm wrong???

**Comment by cjcornell** [comment]
- I don't see a question exactly, about sidebars.
- you wrote:
>> because in sidebars(left or right or both ) we are not going to display any fields to update or lists .. there will be different things to display 
- that is a design statement, on your part.  (and an incorrect one)
- we would indeed want to display fields and format in sidebars
- in fact, see my comments about the profile image, appearing in a sidebar.  And if sidebars are made to work like other profile areas - then this solves a A LOT of other issues in the near future.  
- and if sidebars work like the profile area ... then the "profile photo" in the sidebar also gets a lot simpler ...

**Comment by yasirkhan177** [comment]
first of all sorry for being slow .. I'm not feeling well from last two days .. Bit okay today .. i will make the profile photo in the sidebar (reading dd and fd).. not a big issue .. 

...........................
in fact, see my comments about the profile image, appearing in a sidebar. And if sidebars are made to work like other profile areas - then this solves a A LOT of other issues in the near future. <<< this was your comment .. right now .. i will simple display data like dd-tab_name ... i will investigate if not lot of amendments are needed to make it work like other profile areas .. then i will do it otherwise we can do it in next milestone. Actually the problem is not about displaying data.. The real problem is to display them in a nice format .. which i'm taking care right now . 

i will write more details once i complete this .. but, don't worry .. it will be highly flexible ..

**Comment by yasirkhan177** [comment]
I'm not going to code straight forward( which is very easy ) .. the reason why i'm taking time is to make it as flexible as possible ..

**Comment by cjcornell** [comment]
OK - no problem...
just read my earlier comments about handing the sidebars very very similar
(if not, identical) to the other tab areas... this would then make handling
the profile photos incredibly simple.
